File: jtech-tasklist-backend/src/main/java/br/com/jtech/tasklist/domain/model/Task.java
package br.com.jtech.tasklist.domain.model;

import main.java.br.com.jtech.tasklist.domain.exception.InvalidTaskException;

import java.time.LocalDateTime;
import java.util.UUID;

public class Task {

    private UUID id;
    private String title;
    private String description;
    private TaskStatus status;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    public Task(String title, String description, LocalDateTime now) {
        validateTitle(title);

        this.id = UUID.randomUUID();
        this.title = title;
        this.description = description;
        this.status = TaskStatus.PENDING;
        this.createdAt = now;
    }

    public void updateDetails(String title, String description, LocalDateTime now) {
        validateTitle(title);

        this.title = title;
        this.description = description;
        this.updatedAt = now;
    }

    public void changeStatus(TaskStatus newStatus, LocalDateTime now) {
        if (this.status == TaskStatus.COMPLETED) {
            throw InvalidTaskException.taskAlreadyCompleted();
        }
        this.status = newStatus;
        this.updatedAt = now;
    }

    private void validateTitle(String title) {
        if (title == null || title.isBlank()) {
            throw InvalidTaskException.titleIsRequired();
        }
    }
}

File: jtech-tasklist-frontend/src/types/Task.ts
export type TaskStatus = "PENDING" | "IN_PROGRESS" | "COMPLETED";
export interface Task {
  id: string;
  title: string;
  description: string;
  status: TaskStatus;
  statusDescription: string;
  createdAt: string;
  updatedAt: string;
}

export interface TaskResponse {
  id: string;
  title: string;
  description?: string | null;
  status: TaskStatus;
  statusDescription?: string | null;
  createdAt?: string;
  updatedAt?: string;
}

export interface CreateTaskRequest {
  title: string;
  description: string | null;
}

export interface UpdateTaskRequest {
  title?: string;
  description?: string | null;
  status?: string;
}
